src/include/mpif.h

```diff
@@ -224,7 +224,7 @@
        INTEGER MPI_OP_NULL
        PARAMETER (MPI_OP_NULL=402653184)
        INTEGER MPI_DATATYPE_NULL
-       PARAMETER (MPI_DATATYPE_NULL=z'0c000000')
+       PARAMETER (MPI_DATATYPE_NULL=201326592)
```

Comment:
Another modification approach is `PARAMETER (MPI_DATATYPE_NULL=int(z'0c000000'))`, because `int` is a built-in function, and compilation optimization makes it efficient. This modification might be easily reviewed.

Of course, directly updating to decimal is also good.

Reply:
If it is convenient, subsequent commits can also update the use of similar non-standard syntax in `mpi.f90`:

https://github.com/microsoft/Microsoft-MPI/blob/f2d849ff483b36add9bc2d18c6d7bbc9ce38f4ad/src/include/mpi.f90#L229

Reply:
`gfortran` may throw out the following warning(s):

```sh
mpif.h:56:19:
Warning: Obsolescent feature: Old-style character length at (1)
mpif.h:57:19:
Warning: Obsolescent feature: Old-style character length at (1)
```

The non-standard writing `character*1` should be replaced with the standard writing `character(1)`:

https://github.com/microsoft/Microsoft-MPI/blob/f2d849ff483b36add9bc2d18c6d7bbc9ce38f4ad/src/include/mpi.f90#L58

https://github.com/microsoft/Microsoft-MPI/blob/f2d849ff483b36add9bc2d18c6d7bbc9ce38f4ad/src/include/mpif.h#L56